redfish/provider/common.go

```diff
@@ -88,19 +88,24 @@ func getSystemResource(service *gofish.Service) (*redfish.ComputerSystem, error)
 // NewConfig function creates the needed gofish structs to query the redfish API
 // See https://github.com/stmcginnis/gofish for details. This function returns a Service struct which can then be
 // used to make any required API calls.
-func NewConfig(pconfig *redfishProvider, rserver *models.RedfishServer) (*gofish.Service, error) {
+func NewConfig(pconfig *redfishProvider, rserver *[]models.RedfishServer) (*gofish.Service, error) {
+	if len(*rserver) < 1 {
```

Comment:
Looks like this logic can be added in redfish server schema validator https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.12.0/listvalidator#SizeBetween

Reply:
we've added that validation too at schema level.

Reply:
Thats not true. The schema validator says that no more than 1 server details can be provided, whereas the highlighted line of code validates that no less than 1 server detail is provided.

Reply:
removing the code
To-Do: Verify from plan modifier, if required implement wrapper for validation of unknown in redfish_server.